### PR TITLE
refactor(navigation secondary): improve context-provider use

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1.  Moved context-provider to encapsulate all of the mobile menu contents
2.  Refactored styles to better use context tokens

## Todo

- [x] Add changeset

## Testing Instructions

1. View the Deploy Preview

## Notes to Reviewers

Please check for visual regressions between the DP and ux. on demos for both light and dark navs:

[DP light demo](https://deploy-preview-1108--red-hat-design-system.netlify.app/elements/navigation-secondary/demo/) | [ux dot light demo](https://ux.redhat.com/elements/navigation-secondary/demo/)
[DP dark demo](https://deploy-preview-1108--red-hat-design-system.netlify.app/elements/navigation-secondary/demo/dark-variant/) | [ux dot dark demo](https://ux.redhat.com/elements/navigation-secondary/demo/dark-variant/)

### CODE REVIEW

**Review by zeroedin — COMMENTED** [review]
CTA in dark variant dropdown is now showing dark mode, this is due to being inside the context-provider which makes sense, and is most likely the reason I didn't wrap the navigation prior.  Will have to think about this approach more...

image

**Comment by zeroedin** [comment]
>  Will have to think about this approach more...

Resolved by applying `@colorContextProvider()` to menu to scope it from the parent `rh-context-provider`

File: elements/rh-navigation-secondary/rh-navigation-secondary.css [hunk 1]
```diff
@@ -58,24 +53,19 @@ nav.rtl {
   grid-template-columns: 1fr max-content;
   grid-template-areas:
     "logo menu"
-    "nav nav"
-    "cta cta";
+    "provider provider";
```
Comment (bennypowers) [inline elements/rh-navigation-secondary/rh-navigation-secondary.css:56]: on call we talked about renaming this

options:

- main
- nav
- menu
- mobile menu

anything other than provider, really

**Review by bennypowers — CHANGES_REQUESTED** [review]

File: .changeset/spotty-pots-taste.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@rhds/elements": patch
+---
+
+`<rh-navigation-secondary>`: Refactored CSS and context-provider placement, simplifying CSS.
```
Comment (bennypowers) [inline .changeset/spotty-pots-taste.md:5]:
```suggestion
`<rh-navigation-secondary>`: reduce, validate, and simplify styles
```

File: elements/rh-navigation-secondary/rh-navigation-secondary.css (hunk 1 above)
Comment (zeroedin) [reply elements/rh-navigation-secondary/rh-navigation-secondary.css]: fixed in [01d6710](https://github.com/RedHat-UX/red-hat-design-system/pull/1108/commits/01d67103b133562ff0c9ddcb1c074f0de08d80c1)

File: elements/rh-navigation-secondary/rh-navigation-secondary-menu.ts [hunk 3]
```diff
@@ -30,6 +33,9 @@ import styles from './rh-navigation-secondary-menu.css';
 export class RhNavigationSecondaryMenu extends LitElement {
   static readonly styles = [styles];
 
+  @colorContextProvider()
```
Comment (bennypowers) [inline elements/rh-navigation-secondary/rh-navigation-secondary-menu.ts:36]: doc blocks

**Review by bennypowers — APPROVED** [review]

**Pull request merged**